turbine-core/src/main/java/com/netflix/turbine/discovery/Instance.java

```diff
@@ -74,6 +74,12 @@ public boolean equals(Object obj) {
         equals &= (this.hostname != null) ? (this.hostname.equals(other.hostname)) : (other.hostname == null);
         equals &= (this.cluster != null) ? (this.cluster.equals(other.cluster)) : (other.cluster == null);
         equals &= (this.isUp == other.isUp);
+        if(attributes != null && attributes.get("port") != null){
+        	String port = attributes.get("port");
+        	if(other.getAttributes() != null && other.getAttributes().get("port") != null){
+        		equals &= (port == other.getAttributes().get("port"));
```

Comment:
This code is broken. You are comparing strings with "==". It will always be false.